src/libtiled/maprenderer.cpp

```diff
@@ -259,7 +274,7 @@ void CellRenderer::render(const Cell &cell, const QPointF &pos, const QSizeF &si
     // The USHRT_MAX limit is rather arbitrary but avoids a crash in
     // drawPixmapFragments for a large number of fragments.
     if (mTile != tile || mFragments.size() == USHRT_MAX)
```

Comment:
In theory we need to flush as well when the layer, or at least the tint color, changed. In practice I don't think a `CellRenderer` instance is ever used with different layers so the tint color won't change. As such, I would actually suggest we add the tint color as constructor parameter to the `CellRenderer` instead.

Reply:
done